src/components/editor/CodeEditor.tsx

```diff
@@ -0,0 +1,69 @@
+import 'codemirror/addon/dialog/dialog';
+import 'codemirror/addon/dialog/dialog.css';
+import 'codemirror/addon/hint/anyword-hint';
+import 'codemirror/addon/hint/show-hint';
+import 'codemirror/addon/hint/show-hint.css';
+import 'codemirror/addon/search/jump-to-line';
+import 'codemirror/addon/search/matchesonscrollbar';
+import 'codemirror/addon/search/search';
+import 'codemirror/mode/python/python';
+
+import * as React from 'react';
+import * as ReactCodeMirror from 'react-codemirror';
+import * as _ from 'underscore';
+
+import {CodeMirrorGutters, CodeMirrorModes} from './EditorConstants';
+
+export interface ICodeEditorProps {
+    value: string;
+    readOnly?: boolean;
+    onChange?: (code: string) => void;
+    errorMessage?: string;
+    mode?: string;
+}
+
+export class CodeEditor extends React.Component<ICodeEditorProps> {
+    static defaultProps: Partial<ICodeEditorProps> = {
+        mode: CodeMirrorModes.Python,

```

Comment:
Because JSONEditor has its own component. The only code editor we have that isn't a JSON editor is using python. This way we do not have to set the mode. But I actually think I should not make it optional, this way there won't be any default.